File: lib/entity/pendency_page_entity.dart
import 'pendency_entity.dart';

/// 未决分页实体
class PendencyPageEntity {
  /// 未读数量
  int unreadCnt;

  /// 数据列表
  List<PendencyEntity> items;

  /// 序列号
  int seq;

  /// 时间戳
  int timestamp;

  PendencyPageEntity({this.unreadCnt, this.items, this.seq, this.timestamp});

  PendencyPageEntity.fromJson(Map<String, dynamic> json) {
    unreadCnt = json['unreadCnt'];
    if (json['items'] != null) {
      items = new List<PendencyEntity>();
      (json['items'] as List).forEach((v) {
        items.add(new PendencyEntity.fromJson(v));
      });
    }
    seq = json['seq'];
    timestamp = json['timestamp'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['unreadCnt'] = this.unreadCnt;
    if (this.items != null) {
      data['items'] = this.items.map((v) => v.toJson()).toList();
    }
    data['seq'] = this.seq;
    data['timestamp'] = this.timestamp;
    return data;
  }
}

File: ios/Classes/entity/PendencyPageEntity.swift
import ImSDK;

//  Created by 蒋具宏 on 2020/2/14.
//  未决分页列表
public class PendencyPageEntity : NSObject{
    /**
     *  未读数量
     */
    var unreadCnt : UInt64?;
    
    /**
     *  序列号
     */
    var seq : UInt64?;
    
    
    /**
     *  时间戳
     */
    var timestamp : UInt64?;
    
    
    /**
     *  序列号
     */
    var items : [PendencyEntity]?;
    
    override init() {
    }
    
    init(res : TIMFriendPendencyResponse,items : [PendencyEntity]) {
        super.init();
        self.unreadCnt = res.unreadCnt;
        self.seq = res.seq;
        self.timestamp = res.timestamp;
        self.items = items;
    }
}
